### PR TITLE
chore: draft installing docs

### PR DESCRIPTION
**Pull request by WitoDelnat**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: docs/docs/articles/install/advanced-install.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# Advanced install
+
+A variety of advanced topics to further customize your deployment.
+
+## Organization management
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:5]:
```suggestion
## Organization Management
```

File: docs/docs/articles/install/advanced-install.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+### Bootstrap configuration
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:7]:
```suggestion
### Bootstrap Configuration
```

File: docs/docs/articles/install/advanced-install.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+              default_role: member
+              default_environments:
+                - production_1
+              environments:
+                - name: production_1
+                  default_role: run
+                - name: production_2
+```
+
+Note: The default organization and environment mapping only apply on first sign in. After, you can remove users from environments or change roles thru Testkube UI.
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:59]:
```suggestion
Note: The default organization and environment mapping only apply on first sign in. After, you can remove users from environments or change roles through the Testkube UI.
```

File: docs/docs/articles/install/advanced-install.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+Additionally, by default, Testkube Pro creates a personal organization for every new user. When using default organization and environment configuration, you can turn off personal organizations using the following config:
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:61]:
```suggestion
Additionally, by default, Testkube Pro creates a personal organization for every new user. When using the default organization and environment configuration, you can turn off personal organizations using the following config:
```

File: docs/docs/articles/install/advanced-install.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+# WebSockets ingress:
+annotations:
+  nginx.ingress.kubernetes.io/proxy-read-timeout: "3600"
+  nginx.ingress.kubernetes.io/proxy-send-timeout: "3600"
+```
+
+To use your own ingress controller, reach out to our support team and we’ll gladly investigate your ingress of choice. Alternatively, you can give it a try yourself by deploying Testkube and seeing whether gRPC and WebSockets are working properly.
+
+## Bring your own infra
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:111]:
```suggestion
## Bring Your Own Infra
```

File: docs/docs/articles/install/advanced-install.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+  dex:
+    issuer: <dex issuer url>
+dex:
+  enabled: false
+testkube-cloud-api:
+  api:
+    oauth: {} # check out the `testkube-cloud-api.api.oauth` block in the values.yaml for all available settings
+```
+
+## Air-gapped environments
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:179]:
```suggestion
## Air-gapped Environments
```

File: docs/docs/articles/install/advanced-install.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+### Offline License
+
+By default, Testkube will work with licenses that require internet connectivity. These licenses have the following format: `XXXXXX-XXXXXX-XXXXXX-XXXXXX-XXXXXX-V3`. However, if you want to use Testkube in offline environments you will need to use an offline license.
+
+Contact support if you need an offline license.
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:185]: Should we tell them here how to make contact?

File: docs/docs/articles/install/advanced-install.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+Once you obtained an offline license, you should create a Shared Secret and afterwards
+
+```bash
+global:
+  enterpriseOfflineAccess: true
+  enterpriseLicenseSecretRef: testkube-enterprise-license
+```
+
+### Artifactory and other Registry proxies
```
Comment (jfermi) [inline docs/docs/articles/install/advanced-install.md:195]:
```suggestion
### Artifactory and Other Registry Proxies
```

File: docs/docs/articles/install/cli.mdx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+</TabItem>
+
+<TabItem value="manual" label="Manual Download">
+
+To install Testkube CLI you\'ll need the following tools:
+
+- [Kubectl](https://kubernetes.io/docs/tasks/tools/), Kubernetes command-line tool
+- [Helm](https://helm.sh/)
+
+1. Download the binary for the version and platform of your choice [here](https://github.com/kubeshop/testkube/releases)
```
Comment (jfermi) [inline docs/docs/articles/install/cli.mdx:62]:
```suggestion
1. Download the binary for the version and platform of your choice [here](https://github.com/kubeshop/testkube/releases).
```

File: docs/docs/articles/install/install-with-cli.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
+# Install Testkube with the CLI
+
+The Testkube CLI includes installation commands to help you set up Testkube for your environment. You can choose from one of our built-in configuration profiles (see below) and the CLI will help you with the last-mile configuration to finalise your setup. You can find instructions on how to install the CLI [here][install-cli].
+
+## Deploy the open-source, standalone agent
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-cli.md:5]:
```suggestion
## Deploy the Open-source, Standalone Agent
```

File: docs/docs/articles/install/install-with-cli.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+The standalone-agent is fully open-sourced under a MIT license [on GitHub](https://github.com/kubeshop/testkube).
+
+```
+testkube init
+```
+
+## Deploy an agent that will connect to a control plane
+
+The agent profile installs an agent that will join a control plane running within a different cluster or namespace. The agent acts as a test runner for your organisation’s environment. You can install multiple agents depending
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-cli.md:17]: "Depending" on what?

File: docs/docs/articles/install/install-with-helm.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+You should enable ingress and configure your domain to access Testkube services:
+
+```
+global:
+  ingress:
+    enabled: true
+  domain: "example.com"
+```
+
+By default, the following services will be exposed. You can also choose to override the subdomain for each service
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:66]:
```suggestion
By default, the following services will be exposed. You can also choose to override the subdomain for each service.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+| Status Pages API | status.$domain     | global.statusPagesApiSubdomain |
+| Storage API      | storage.$domain    | global.storageApiSubdomain     |
+
+TLS can either be handled through cert-manager or a manually defined secret. While custom TLS certificates are possible, we strongly recommend using `cert-manager` for simplified certificate manager.
+
+:::info
+TLS should be terminated at the application-level instead of the ingress-level as it will drastically enhance the performance of the gRPC and Websocket protocols. Be mindful that NGINX defaults to a downgrade from HTTP2 to HTTP1.1 when the backend service is using an insecure port.
+:::
+
+#### Custom certificates
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:83]:
```suggestion
#### Custom Certificates
```

File: docs/docs/articles/install/install-with-helm.md [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+
+- `tls.crt` - the certificate
+- `tls.key` - the private key
+- `ca.crt` - the CA certificate (if the certificate is not self-signed)
+
+If certificate-based authentication is required, the custom certificates need to be configured in the following places:
+
+- Enterprise API
+  - If `MINIO_ENDPOINT` is set to an exposed URL, then the following Helm values need to be configured:
+    - The following Helm parameter needs to be enabled to inject the custom certificate into MinIO `testkube-cloud-api.minio.certSecret.enabled: true`
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:95]:
```suggestion
    - The following Helm parameter needs to be enabled to inject the custom certificate into MinIO `testkube-cloud-api.minio.certSecret.enabled: true`.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+    - If the certificate is not self-signed, the CA cert needs to be injected also by enabling the Helm parameter `testkube-cloud-api.minio.mountCACertificate: true`
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:96]:
```suggestion
    - If the certificate is not self-signed, the CA cert needs to be injected also by enabling the Helm parameter `testkube-cloud-api.minio.mountCACertificate: true`.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+    - Custom certificate verification can also be skipped by setting `testkube-cloud-api.minio.skipVerify: true`
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:97]:
```suggestion
    - Custom certificate verification can also be skipped by setting `testkube-cloud-api.minio.skipVerify: true`.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+  - If `MINIO_ENDPOINT` uses the Kubernetes DNS record (`testkube-enterprise-minio.<namespace>.svc.cluster.local:9000`), `AGENT_STORAGE_HOSTNAME` should be set to point to the exposed storage URL
+- Agent
+  - Agent API
+    - If the Enterprise API is configured to use certificate-based authentication or is using a certificate signed by a custom CA, the Agent API needs to be configured to use the same certificates by pointing `testkube-api.cloud.tls.certificate.secretRef` to the Kubernetes secret which contains the certificates
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:101]:
```suggestion
    - If the Enterprise API is configured to use certificate-based authentication or is using a certificate signed by a custom CA, the Agent API needs to be configured to use the same certificates by pointing `testkube-api.cloud.tls.certificate.secretRef` to the Kubernetes secret which contains the certificates.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+    - Custom certificate verification can also be skipped by setting `testkube-api.cloud.tls.skipVerify: true`
+  - Storage
+    - The following Helm parameter needs to be enabled to inject the custom certificate into MinIO `testkube-api.storage.certSecret.enabled: true`
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:104]:
```suggestion
    - The following Helm parameter needs to be enabled to inject the custom certificate into MinIO `testkube-api.storage.certSecret.enabled: true`.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+    - If the certificate is not self-signed, the CA cert needs to be injected also by enabling the Helm parameter `testkube-cloud-api.minio.mountCACertificate: true`
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:105]:
```suggestion
    - If the certificate is not self-signed, the CA cert needs to be injected also by enabling the Helm parameter `testkube-cloud-api.minio.mountCACertificate: true`.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 20]
```diff
@@ -0,0 +1,148 @@
...
+    - Custom certificate verification can also be skipped by setting `testkube-api.storage.skipVerify: true`
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:106]:
```suggestion
    - Custom certificate verification can also be skipped by setting `testkube-api.storage.skipVerify: true`.
```

File: docs/docs/articles/install/install-with-helm.md [hunk 21]
```diff
@@ -0,0 +1,148 @@
...
+
+Testkube requires a variety of secrets to operate. Any required secret that is not provided manually will be automatically generated. You can optionally choose to specify your own secrets. For the current shared secrets, it’s recommended to use the autogeneration.
+
+- testkube-default-agent-token
+- testkube-minio-credentials
+- testkube-license
+
+[license]: https://testkube.io/download
+
+## Advanced settings
```
Comment (jfermi) [inline docs/docs/articles/install/install-with-helm.md:144]:
```suggestion
## Advanced Settings
```

File: docs/docs/articles/install/overview.md [hunk 22]
```diff
@@ -0,0 +1,22 @@
+# About installing Testkube
```
Comment (jfermi) [inline docs/docs/articles/install/overview.md:1]:
```suggestion
# About Installing Testkube
```

File: docs/docs/articles/install/quickstart-install.mdx [hunk 23]
```diff
@@ -0,0 +1,117 @@
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Testkube quickstart
```
Comment (jfermi) [inline docs/docs/articles/install/quickstart-install.mdx:4]:
```suggestion
# Testkube Quickstart
```

File: docs/docs/articles/install/quickstart-install.mdx [hunk 24]
```diff
@@ -0,0 +1,117 @@
...
+
+```bash
+curl -sSLf https://get.testkube.io | sh
+```
+
+</TabItem>
+
+<TabItem value="manual" label="Manual Download">
+
+To install Testkube CLI you\'ll need the following tools:
```
Comment (jfermi) [inline docs/docs/articles/install/quickstart-install.mdx:75]:
```suggestion
To install Testkube CLI you'll need the following tools:
```

File: docs/docs/articles/install/quickstart-install.mdx [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+
+- [Kubectl](https://kubernetes.io/docs/tasks/tools/), Kubernetes command-line tool
+- [Helm](https://helm.sh/)
+
+1. Download the binary for the version and platform of your choice [here](https://github.com/kubeshop/testkube/releases)
```
Comment (jfermi) [inline docs/docs/articles/install/quickstart-install.mdx:80]:
```suggestion
1. Download the binary for the version and platform of your choice [here](https://github.com/kubeshop/testkube/releases).
```

File: docs/docs/articles/install/quickstart-install.mdx [hunk 26]
```diff
@@ -0,0 +1,117 @@
...
+2. Unpack it. For example, in Linux use `tar -zxvf testkube_1.5.1_Linux_x86_64.tar.gz`
```
Comment (jfermi) [inline docs/docs/articles/install/quickstart-install.mdx:81]:
```suggestion
2. Unpack it. For example, in Linux use `tar -zxvf testkube_1.5.1_Linux_x86_64.tar.gz`.
```

File: docs/docs/articles/install/quickstart-install.mdx [hunk 27]
```diff
@@ -0,0 +1,117 @@
...
+For Windows, you will need to unpack the binary and add it to the `%PATH%` as well.
+
+:::note
+If you use a package manager that we don\'t support, please let us know here [#161](https://github.com/kubeshop/testkube/issues/161).
+:::
+
+</TabItem>
+</Tabs>
+
+## Run the Testkube installer
```
Comment (jfermi) [inline docs/docs/articles/install/quickstart-install.mdx:97]:
```suggestion
## Run the Testkube Installer
```

File: docs/docs/articles/install/reference-architectures.md [hunk 28]
```diff
@@ -0,0 +1,47 @@
+# Reference Architectures
+
+Testkube contains two sub-systems:
+
+- A Control Plane which includes the Dashboard, Storage for Results/Artifacts, Cluster Federation, etc
+- An Agent running in your cluster that manages Testkube resources, runs tests, gathers results, etc.
+
+The two sub-systems can be deployed in variety of ways to give flexible into where your tests are executed and where test logs and artifacts are stored.
```
Comment (jfermi) [inline docs/docs/articles/install/reference-architectures.md:8]:
```suggestion
The two sub-systems can be deployed in variety of ways to give flexibility into where your tests are executed and where test logs and artifacts are stored.
```

File: docs/docs/articles/install/uninstall.md [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+
+### Uninstalling with Helm
+
+You can use Helm to uninstall Testkube:
+
+```bash
+helm uninstall testkube
+```
+
+### Giving feedback
```
Comment (jfermi) [inline docs/docs/articles/install/uninstall.md:27]:
```suggestion
### Giving Feedback
```

File: docs/docs/articles/install/uninstall.md [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+
+We’re sad to see you going. Did Testkube not meet your expectations? We’re keen on feedback, so feel free to share your experience and we will look into making Testkube better for everyone.
```
Comment (jfermi) [inline docs/docs/articles/install/uninstall.md:29]: Add a link for how to contact.

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

**Comment by WitoDelnat** [comment]
Thanks for the input @jfermi 💪

**Review by jmorante-ks — APPROVED** [review]

**Pull request merged**